bindings/node/lib/clientEncryption.js

```diff
@@ -103,24 +104,22 @@ module.exports = function (modules) {
       this._bson = options.bson || BSON || client.topology.bson;
       this._proxyOptions = options.proxyOptions;
       this._tlsOptions = options.tlsOptions;
+      this._kmsProviders = options.kmsProviders || {};
 
       if (options.keyVaultNamespace == null) {
         throw new TypeError('Missing required option `keyVaultNamespace`');
       }
 
-      Object.assign(options, { cryptoCallbacks });
+      const mongoCryptOptions = { ...options, cryptoCallbacks };
```

Comment:
This is previously mutating the user provided options which could lead to unexpected behaviour.